src/app/services/fractalis.service.ts

```diff
@@ -157,9 +175,30 @@ export class FractalisService {
     });
   }
 
-  setSubsets() {
-    // TODO make use of this function
-    // this.F.setSubsets([])
+  private configSubsetsUpdate() {
+    this.cohortService.cohortsUpdated.asObservable()
+      .subscribe((cohorts: Cohort[]) => {
+        let subjectCalls = [];
+        cohorts
+          .filter((cohort: Cohort) => {
+            return cohort.selected;
+          })
+          .forEach((cohort: Cohort) => {
+            subjectCalls.push(this.resourceService.getSubjects(cohort.constraint));
```

Comment:
Does this fetch patient ids for all cohorts whenever one of the cohorts has changed or selected?

Reply:
yes, whenever one of the cohorts is changed, the cohortsUpdated subject will emit event to inform its subscribers, in this case, the only subscriber here in fractalis service, catches it and does subsets update